### ISSUE TITLE
A community perspective on API packages: challenges and best practices

### ISSUE THREAD

**Issue by Rekyt** [ropensci-org/community-calls#27]
# Topic

### Who is the audience?

Many people from the rOpenSci community develop and maintain API packages.

[r-pkg.org](https://www.r-pkg.org/) references 900+ packages when searching for "API client" https://www.r-pkg.org/search.html?q=API+client

### Why is this important?

 APIs are everywhere nowadays and package developers often face the question of the relevance of developing custom packages for each API compared to "just" using standard tools like `httr`. Despite the number of API packages, few resources exist regarding the specific challenges associated with their development.
rOpenSci now hosts at least 46 API packages https://github.com/search?q=org%3Aropensci+API+language%3AR&type=Repositories
this represents a huge amount of experience in the development of API packages. This experience could be shared to foster better practices and warn future developers about specific issues related to APIs.

### What should be covered?

Challenges to developing and maintaining API packages (scaffolding code, testing, communicating with the API developers).

When TO and NOTTO develop an API package.

Development stories (successes and failures?) of these packages.

### Suggested speakers or contributors

This idea was co-suggested by @maelle and @Bisaloo.
An interesting speaker could be @sckott who developed so many API packages over the years.
Basically any developer who contributed an API package to rOpenSci.

### Resources you would recommend to the audience

A future article on rOpenSci blog: https://github.com/ropensci/roweb3/pull/345

List of resources from the draft article:
- The [R packages book by Hadley Wickham and Jenny Bryan](https://r-pkgs.org/) (the online version corresponds to the future second edition, not completely revised at the time of writing);
- [httr2 "Wrapping APIs" vignette](https://httr2.r-lib.org/articles/wrapping-apis.html) (or httr vignettes, or curl docs, depending on what dependency you prefer);
- [HTTP testing in R](https://books.ropensci.org/http-testing/) including a more general chapter on ["Graceful HTTP R packages"](https://books.ropensci.org/http-testing/graceful.html).
